components/script/dom/xmlhttprequest.rs

```diff
@@ -797,6 +797,8 @@ impl XMLHttpRequest {
             }
         }
 
+        *self.response_url.borrow_mut() = metadata.final_url.serialize_no_fragment();
```

Comment:
I'm using `metadata.final_url` right now, but the wpt redirection test seems to be failing again.